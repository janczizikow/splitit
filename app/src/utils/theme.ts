const colors = {
  primary: "#00ccbb",
  error: "#ff483b",
  errorBg: "#ffebe4",
  success: "#006d68",
  successBg: "#edfad0",
  border: "#bac3c3",
  greyBg: "#f9fafa",
  lightGreyBg: "#f5f5f5",
  headingColor: "#2e3333"
};

const borderRadius = 2;

export interface Theme {
  colors: typeof colors;
  borderRadius: number;
}
export default {
  colors,
  borderRadius
};
