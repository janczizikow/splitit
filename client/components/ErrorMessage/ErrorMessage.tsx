import React from "react";
import styled from "@emotion/styled";
import { ErrorResponse } from "apollo-link-error";

const ErrorMessage = styled.div`
  padding: 1rem;
  margin-bottom: 1.5rem;
  font-size: 0.875rem;
  color: ${p => p.theme.colors.error};
  text-align: center;
  background-color: ${p => p.theme.colors.errorBg};
`;

interface Props {
  error?: {} | ErrorResponse;
}

const DisplayError: React.FunctionComponent<Props> = ({ error }) => {
  if (!error || !error.message) return null;
  if (
    error.networkError &&
    error.networkError.result &&
    error.networkError.result.errors.length
  ) {
    return error.networkError.result.errors.map((error, i: number) => (
      <ErrorMessage key={i}>
        <p>
          <strong>Error</strong> {error.message.replace("GraphQL error: ", "")}
        </p>
      </ErrorMessage>
    ));
  }
  return (
    <ErrorMessage>
      <p>
        <strong>Error!</strong> {error.message.replace("GraphQL error: ", "")}
      </p>
    </ErrorMessage>
  );
};

ErrorMessage.displayName = "ErrorMessage";
DisplayError.defaultProps = {
  error: {}
};

export default DisplayError;
