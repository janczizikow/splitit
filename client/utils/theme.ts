const colors = {
  primary: "#00ccbb",
  error: "#ff483b",
  errorBg: "#ffebe4",
  success: "#006d68",
  successBg: "#edfad0",
  border: "#bac3c3",
  greyBg: "#f9fafa",
  lightGreyBg: "#f5f5f5",
  headingColor: "#2e3333"
};

const borderRadius = 2;

const gridGutters = 16;
const gridColumns = 12;
const gridBreakpoints = [0, 576, 768, 992, 1200];

export interface Theme {
  colors: typeof colors;
  borderRadius: number;
  gridGutters: number | string;
  gridColumns: number;
  gridBreakpoints: number[];
}

export default {
  colors,
  borderRadius,
  gridGutters,
  gridColumns,
  gridBreakpoints
};
