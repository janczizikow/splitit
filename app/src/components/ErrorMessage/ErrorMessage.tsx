import React from "react";
import { View, StyleSheet } from "react-native";
import { Text } from "react-native-elements";
import { ErrorResponse } from "apollo-link-error";
import theme from "../../utils/theme";

interface Props {
  error?: {} | ErrorResponse;
}

const ErrorMessage: React.FunctionComponent<Props> = ({ error }) => {
  if (!error || !error.message) return null;
  if (
    error.networkError &&
    error.networkError.result &&
    error.networkError.result.errors.length
  ) {
    return error.networkError.result.errors.map((error, i: number) => (
      <View key={i} style={styles.container}>
        <Text style={styles.errorText}>
          <Text style={styles.strong}>Error</Text>{" "}
          {error.message.replace("GraphQL error: ", "")}
        </Text>
      </View>
    ));
  }
  return (
    <View style={styles.container}>
      <Text style={styles.errorText}>
        <Text style={styles.strong}>Error!</Text>{" "}
        {error.message.replace("GraphQL error: ", "")}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    marginHorizontal: 12,
    marginBottom: 24,
    textAlign: "center",
    backgroundColor: theme.colors.errorBg
  },
  errorText: {
    color: theme.colors.error
  },
  strong: {
    fontWeight: "bold"
  }
});

export default ErrorMessage;
