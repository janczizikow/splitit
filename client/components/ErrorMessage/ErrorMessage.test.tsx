import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ErrorMessage from "./ErrorMessage";
import theme from "../../utils/theme";

describe("<ErrorMessage/>", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<ErrorMessage />);
    expect(wrapper).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const wrapper = mount(<ErrorMessage theme={theme} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
