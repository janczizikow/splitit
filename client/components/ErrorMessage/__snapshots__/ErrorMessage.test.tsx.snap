// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage/> matches snapshot 1`] = `
<DisplayError
  error={Object {}}
  theme={
    Object {
      "borderRadius": 2,
      "colors": Object {
        "border": "#bac3c3",
        "error": "#ff483b",
        "errorBg": "#ffebe4",
        "greyBg": "#f9fafa",
        "headingColor": "#2e3333",
        "lightGreyBg": "#f5f5f5",
        "primary": "#00ccbb",
        "success": "#006d68",
        "successBg": "#edfad0",
      },
      "gridBreakpoints": Array [
        0,
        576,
        768,
        992,
        1200,
      ],
      "gridColumns": 12,
      "gridGutters": 16,
    }
  }
/>
`;
